pallets/marketplace/src/lib.rs

```diff
@@ -908,10 +1003,21 @@ pub mod pallet {
                             Ok(())
                         },
                     )?;
-                    <StoredMatchesReverseIndex<T>>::insert(
+                    <AssignedProcessors<T>>::try_mutate(
                         &m.job_id,
-                        planned_execution.source.clone(),
-                    );
+                        |processor| -> Result<(), Error<T>> {
+                            match processor {
+                                None => {
+                                    *processor = Some(Default::default());
+                                }
+                                Some(p) => {
+                                    p.try_push(planned_execution.source.clone())
+                                        .map_err(|_| Error::<T>::TooManySlots)?;
```

Comment:
How does this mean `TooManySlots`?

```suggestion
                                    p.try_push(planned_execution.source.clone())
                                        .map_err(|_| Error::<T>::TooManySlots)?;
```

Reply:
according to docs of `try_push`, pushing to the end of a BoundedVec only fails if the length is exceeded. This error cannot happen here since we validate slots before, but we have to handle the error

Reply:
replaced it with StorageDoubleMap